core/domain/exp_services_test.py

```diff
@@ -7625,6 +7627,395 @@ def test_revert_exploration_with_mismatch_of_versions_raises_error(
+                                'html': '<p>1</p>'
+                            },
+                            {
+                                'content_id': 'ca_choices_6',
+                                'html': '<p>2</p>'
+                            },
+                            {
+                                'content_id': 'ca_choices_7',
+                                'html': '<p>3</p>'
+                            },
+                            {
+                                'content_id': 'ca_choices_8',
+                                'html': '<p>4</p>'
+                            }
+                        ]
+                    },
+                    'showChoicesInShuffledOrder': {
+                        'value': False
+                    },
+                    'maxAllowableSelectionCount': {
+                        'value': 1
+                    },
+                    'minAllowableSelectionCount': {
+                        'value': 1
+                    }
+                }
+            }),
+            exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                'state_name': 'Introduction',
+                'property_name': 'answer_groups',
+                'new_value': [{
+                    'outcome': { 
+                        'dest': 'End',
+                        'dest_if_really_stuck': None,
+                        'param_changes': [],
+                        'refresher_exploration_id': None,
+                        'missing_prerequisite_skill_id': None,
+                        'feedback': {
+                            'html': '<p>correct</p>',
+                            'content_id': 'feedback_9'},
+                        'labelled_as_correct': True
+                    },
+                    'rule_specs': [{
+                        'rule_type': 'Equals',
+                        'inputs': {
+                            'x': 0
+                        }
+                    }],
+                    'training_data': [],
+                    'tagged_skill_misconception_id': None
+                }],
+            }),
+            exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_EDIT_EXPLORATION_PROPERTY,
+                'new_value': 10,
+                'old_value': 9,
+                'property_name': 'next_content_id_index'
+            })]
+
+        exp_services.update_exploration(
+            self.owner_id, self.EXP_0_ID, change_list, 'Update 1')
+
+        entity_translation_models: Sequence[
+            translation_models.EntityTranslationsModel
+        ] = (
+            translation_models.EntityTranslationsModel.get_all().fetch())
+        exp_opportunity_summary: Optional[
+            opportunity_domain.ExplorationOpportunitySummary] = (
+                opportunity_services.get_exploration_opportunity_summary_by_id(
+                    self.EXP_0_ID))
+        if exp_opportunity_summary is not None:
```

Comment:
Why if condition in the test?

Reply:
`get_exploration_opportunity_summary_by_id()` returns an `Optional[ExplorationOpportunitySummary]`. So, in this case if we do not check `exp_opportunity_summary` to be not none then `exp_opportunity_summary.translation_counts` would throw an mypy type error.